### PR TITLE
fix valigrind identified memory leaks in cmd-ls.c

### PR DESCRIPTION
**Pull request by kyleburton**
cmake -DCMAKE_BUILD_TYPE=Debug .
make
valgrind --log-file=./valgrind.log --leak-check=full ./lpass ls --sync=now

Signed-off-by: Kyle Burton <kyle.burton@gmail.com>

### CODE REVIEW

File: cmd-ls.c [hunk 1]
```diff
@@ -373,11 +373,14 @@ int cmd_ls(int argc, char **argv)
 	session_free(session);
 	blob_free(blob);
 
+  /* NB: valgrind identified something related to account_array as a leak, though
+   * freeing the entries crashes the cli.
```
Comment (bcopeland) [inline cmd-ls.c]: The reason this crashes is that most entries in the account_array[i] are just a copy of the account already stored in blob's account list; so the account freeing that happens in blob_free() ends up as a double free.

The leak is that _some_ of the accounts in account_array (the shares) are bogus creations just so qsort() will do something sensible.  So, we need to identify those and free only those.

Comment (kyleburton) [inline cmd-ls.c]: Yes, please note that this PR does resolve a leak in endpoints.c as well.

File: endpoints.c [hunk 2]
```diff
@@ -248,6 +248,8 @@ void lastpass_log_access(enum blobsync sync, const struct session *session, unsi
 		http_post_add_params(&params, "sharedfolderid", account->share->id, NULL);
 
 	upload_queue_enqueue(sync, key, session, "loglogin.php", &params);
+
+  free(params.argv);
```
Comment (bcopeland) [inline endpoints.c]: Thanks, I cherry-picked this one, fixed up the whitespace and edited commit log a bit.

**Comment by bcopeland** [comment]
Thanks! Care to respin the cmd-ls fix?

**Comment by kyleburton** [comment]
I'll take another shot at it.  Is there a way to run off-line tests with the cli?  Can we create a blob fixture w/some representative data so integration style tests can be written?  That'd help automate things like the valgrind analysis and perhaps facilitate trying out AFL on the code (this as an opportunity for me to try out something like AFL).

**Comment by bcopeland** [comment]
Yes, look in the test directory.

I have run AFL before, but the tricky thing there is the blob itself, being encrypted, doesn't lend itself well to fuzzing (most attempts just result in an undecryptable blob).  So you have to hack it to save the blob in plaintext if you want to use the blob as a test input.

**Pull request closed without merging**